### ISSUE TITLE
Can't pickle unittest.TestCase instances

### ISSUE THREAD

**Issue by GoogleCodeExporter** [testing-cabal/unittest-ext#42]
```
Taken from: http://bugs.python.org/issue10326


In Python 2.7 a change was introduced to TestCase which involves storing a 
dictionary of method objects on TestCase instances. This makes them 
unpickleable.

unittest2 stores strings (method names) instead of method objects (a fix to 
make TestCase instances copyable under earlier versions of Python). The same 
fix could be applied to unittest.
```

Original issue reported on code.google.com by `MarkRo...@gmail.com` on 17 Nov 2010 at 1:40

**Comment by GoogleCodeExporter** [comment]
```
This issue is not present in unittest2 or unittest2-py3k, but the attached 
patches port the test fixtures which check for the issue.
```

Original comment by `MarkRo...@gmail.com` on 17 Nov 2010 at 1:41

Attachments:
- [unittest2.pickle.patch](https://storage.googleapis.com/google-code-attachments/unittest-ext/issue-42/comment-1/unittest2.pickle.patch)
- [unittest2py3k.pickle.patch](https://storage.googleapis.com/google-code-attachments/unittest-ext/issue-42/comment-1/unittest2py3k.pickle.patch)

**Comment by GoogleCodeExporter** [comment]
```
Fixed in unittest2. Needs fixing in unittest2-py3k and Python 2.7.
```

Original comment by `fuzzyman` on 20 Nov 2010 at 5:24

**Comment by GoogleCodeExporter** [comment]
```
Open issue on Python for porting fix to 2.7. unittest2-py3k should be changed 
to a patch against Python 3 trunk so fixes applied to Python 3 won't need 
specially porting to unittest2-py3k.
```

Original comment by `fuzzyman` on 24 Nov 2010 at 12:37
- Changed state: **Fixed**
